pkg/pillar/cmd/client/client.go

```diff
@@ -634,6 +636,11 @@ func handleDNSModify(ctxArg interface{}, key string, statusArg interface{}) {
 		// ledmanager subscribes to DeviceNetworkStatus to see changes
 		ctx.usableAddressCount = newAddrCount
 	}
+
+	// update proxy certs if configured
+	if ctx.zedcloudCtx != nil && ctx.zedcloudCtx.V2API {
```

Comment:
I wonder whether we want to have a function which can check whether any proxy certs changed, and only do this update if there is a change. More of an issue for the long-running agents like zedagent and logmanager than client.

Reply:
I was thinking on this direction. Although the AppendCertFromPEM() function I found does check for duplication, so it's very light weight. This may save us from this bookkeeping.